addons/account/models/account_move_line.py

```diff
@@ -1547,13 +1550,22 @@ def changed(fname):
                     line.balance = amount_currency
 
         after = existing()
+        changed_currency_rate = any(changed_existing_val('currency_rate') for line in after)
+        term_line = False
+        aggregated_balance = 0
         for line in after:
+            if line.display_type == 'payment_term' and changed_currency_rate and len(after) > 1:
+                term_line = line
+                continue
             if (
                 (changed('amount_currency') or changed('currency_rate') or changed('move_type'))
                 and (not changed('balance') or (line not in before and not line.balance))
             ):
                 balance = line.company_id.currency_id.round(line.amount_currency / line.currency_rate)
                 line.balance = balance
+            aggregated_balance += abs(line.balance)
```

Comment:
Why the `abs`? Isn't it possible to have some negative and positive lines at the same time? Wouldn't that make the calculation incorrect?

Reply:
You are absolutely correct. Having it without the `abs` was breaking some tests but it was due to some faulty logic elsewhere which has now been fixed!